### PR TITLE
refactor(eagle3): align draft model module names with sglang

### PR DESCRIPTION
**Pull request by khazic**
# What does this PR do ?

Rename submodules and restructure the EAGLE-3 Llama draft model so the saved state dict matches SGLang's `LlamaForCausalLMEagle3` (`python/sglang/srt/models/llama_eagle3.py`) exactly. With this alignment, a checkpoint produced by `train_eagle3.py` can be loaded by SGLang for inference with **zero key remapping** — `load_weights` fuses `q/k/v_proj → qkv_proj` and `gate/up_proj → gate_up_proj` via the standard `stacked_params_mapping`.

# Changelog

- `nemo_automodel/components/speculative/eagle/draft_llama.py`:
  - Introduce inner `Eagle3LlamaModel` backbone holding `embed_tokens / fc / layers (ModuleList[1]) / norm`.
  - Rename `self.decoder` → `self.model.layers[0]`, `self.hidden_proj` → `self.model.fc`, `self.embed_tokens` → `self.model.embed_tokens`, `self.norm` → `self.model.norm`.
  - Rename `Eagle3LlamaDecoderLayer.input_emb_layernorm` → `input_layernorm`, `hidden_layernorm` → `hidden_norm`.
  - Add `layer_id` / `is_input_layer` flags on the decoder layer.
  - Source `fc` input dim from `config.num_aux_hidden_states` (default 3, matches previous hardcoded value).
  - Set `base_model_prefix = "model"`.
- `tests/unit_tests/speculative/test_eagle3.py`: update attribute accesses to the new names (`draft.model.layers[0].self_attn`, `layer.input_layernorm`, `layer.hidden_norm`).
- `nemo_automodel/recipes/llm/train_eagle3.py`: update an inline comment that referenced the old `hidden_proj` name.

**Resulting state-dict keys** (drop-in compatible with `sglang.srt.models.llama_eagle3.LlamaForCausalLMEagle3.load_weights`):

```
model.embed_tokens.weight
model.fc.weight
model.layers.0.input_layernorm.weight
model.layers.0.hidden_norm.weight
model.layers.0.post_attention_layernorm.weight
model.layers.0.self_attn.{q,k,v,o}_proj.weight
model.layers.0.mlp.{gate,up,down}_proj.weight
model.norm.weight
lm_head.weight
```

# Before your PR is "Ready for review"

**Pre checks**:

- [x] Read and followed [Contributor guidelines](https://github.com/NVIDIA-NeMo/Automodel/blob/main/CONTRIBUTING.md)
- [x] All 14 existing unit tests in `tests/unit_tests/speculative/test_eagle3.py` pass.
- [x] Verified bit-exact equivalence (`max_diff = 0.0`) against the previous implementation on CPU/fp32 for `project_hidden_states`, single-step forward, 3-step TTT forward, and `compute_logits` — this is a pure-rename refactor.
- [x] DCO sign-off included.

# Additional Information

This is a structural rename only; no math or training behavior changes. Verified via:
- All existing unit tests pass (14/14).
- Side-by-side comparison of `HEAD~1` vs `HEAD` with identical seed and remapped weights: bit-exact output across forward, TTT cache path, and logits.

### CODE REVIEW

**Comment by khazic** [comment]
Still running a training pass on the same dataset I've been using; once done I'll feed the resulting checkpoint directly into SGLang for inference (the rename in this PR makes it a drop-in load — no key remapping needed). I'll post the training logs and the SGLang inference outputs in this thread before we merge.

The motivation for this change is exactly that direct-inference path: with module names perfectly aligned to `sglang/srt/models/llama_eagle3.py`, the trainer's `draft_model.pt` can be served by SGLang as-is.

**Review by HuiyingLi — APPROVED** [review]
Thank you so much!
